src/messaging/dotnet/src/Client/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -36,8 +36,9 @@ public static IServiceCollection AddMessageRouter(
     {
         var builder = new MessageRouterBuilder(serviceCollection);
         builderAction(builder);
-        serviceCollection.AddSingleton<IMessagingService, MessageRouterClient>();
+
         serviceCollection.AddSingleton<IMessageRouter, MessageRouterClient>();
+        serviceCollection.AddTransient<IMessagingService>(p => p.GetRequiredService<IMessageRouter>());
```

Comment:
Can you please explain this change?

Reply:
In the original version MessageRouterClient exists in 2 different singleton instances while it should be a single one - and this is the way to do that.